clarity/evaluator/haspi/eb.py

```diff
@@ -1457,176 +1792,247 @@ def bm_covary(xBM, yBM, segsize, fsamp):
+        lp_filter_order = np.array([1, 3, 5, 5, 5, 5])  # LP filter order
+    if freq_cutoff is None:
+        freq_cutoff = 1000 * np.array(
+            [1.5, 2.0, 2.5, 3.0, 3.5, 4.0]
+        )  # Cutoff frequencies in Hz
+    fsync = np.zeros((6, n_channels))  # Array of filter freq resp vs band center freq
     for n in range(6):
-        fc2p = fcut[n] ** (2 * p[n])
-        freq2p = cfreq ** (2 * p[n])
+        fc2p = freq_cutoff[n] ** (2 * lp_filter_order[n])
+        freq2p = _center_freq ** (2 * lp_filter_order[n])
         fsync[n, :] = np.sqrt(fc2p / (fc2p + freq2p))
 
     # Find the segments that lie sufficiently above the threshold.
-    sigRMS = np.sqrt(sigMSx)  # Convert squared amplitude to dB envelope
-    sigLinear = 10 ** (sigRMS / 20)  # Linear amplitude (specific loudness)
-    xsum = np.sum(sigLinear, 0) / nchan  # Intensity averaged over frequency bands
-    xsum = 20 * np.log10(xsum)  # Convert back to dB (loudness in phons)
-    index = np.argwhere(xsum > thr).T  # Identify those segments above threshold
+    signal_rms = np.sqrt(
+        reference_signal_mean_square
+    )  # Convert squared amplitude to dB envelope
+    signal_linear_amplitude = 10 ** (
+        signal_rms / 20
+    )  # Linear amplitude (specific loudness)
+    reference_mean = (
+        np.sum(signal_linear_amplitude, 0) / n_channels
+    )  # Intensity averaged over frequency bands
+    reference_mean = 20 * np.log10(
+        reference_mean
+    )  # Convert back to dB (loudness in phons)
+    index = np.argwhere(
+        reference_mean > threshold_db
+    ).T  # Identify those segments above threshold
     if index.size != 1:
         index = index.squeeze()
     nseg = index.shape[0]  # Number of segments above threshold
 
     # Exit if not enough segments above zero
     if nseg <= 1:
         print("Function eb.AveCovary: Ave signal below threshold, outputs set to 0.")
-        avecov = 0
+        average_covariance = 0
         # syncov = 0
-        syncov = [0] * 6
-        return avecov, syncov
+        ihc_sync_covariance = [0] * 6
+        return average_covariance, ihc_sync_covariance
 
     # Remove the silent segments
-    sigcov = sigcov[:, index]
-    sigRMS = sigRMS[:, index]
+    signal_cross_covariance = signal_cross_covariance[:, index]
+    signal_rms = signal_rms[:, index]
 
     # Compute the time-frequency weights. The weight=1 if a segment in a
     # frequency band is above threshold, and weight=0 if below threshold.
-    weight = np.zeros((nchan, nseg))  # No IHC synchronization roll-off
-    wsync1 = np.zeros((nchan, nseg))  # Loss of IHC synchronization at high frequencies
-    wsync2 = np.zeros((nchan, nseg))
-    wsync3 = np.zeros((nchan, nseg))
-    wsync4 = np.zeros((nchan, nseg))
-    wsync5 = np.zeros((nchan, nseg))
-    wsync6 = np.zeros((nchan, nseg))
-    for k in range(nchan):
+    weight = np.zeros((n_channels, nseg))  # No IHC synchronization roll-off
+    wsync1 = np.zeros(
```

Comment:
This would all be much nicer if the original code had wsync also as an array, but suggest we leave that for now.

Reply:
Captured as #154 I did notice this sort of structure in at least one other place and refactored to using an array, missed this one though.